app/dossierPrograms/dossierPrograms.controllers.js

```diff
@@ -206,6 +206,202 @@ dossierProgramsModule.controller("dossiersProgramSectionController", [
+                    grandParentOrganisationUnit: false,
+                },
+
+                axes: [],
+                translations: [],
+                yearlySeries: [],
+                interpretations: [],
+                userGroupAccesses: [
+                    {
+                        access: "rw------",
+                        userGroupUid: "epFY01iJN0Z",
+                        displayName: "ALL USERS",
+                        id: "epFY01iJN0Z",
+                    },
+                ],
+                subscribers: [],
+                userAccesses: [],
+                favorites: [],
+                columns: [
+                    {
+                        dimension: "pe",
+                        items: [
+                            {
+                                id: "THIS_YEAR",
+                            },
+                        ],
+                    },
+                    {
+                        dimension: "BtFXTpKRl6n",
+                        items: [
+                            {
+                                id: "ALL_ITEMS",
+                            },
+                        ],
+                    },
+                ],
+                filters: [
+                    {
+                        dimension: "ou",
+                        items: [
+                            {
+                                id: "USER_ORGUNIT",
+                            },
+                        ],
+                    },
+                ],
+                rows: [
+                    {
+                        dimension: "dx",
+                        items: [
+                            {
+                                id: "DqqSJFWB392",
+                            },
+                            {
+                                id: "qywsusOdy33",
+                            },
+                        ],
+                    },
+                ],
+                series: [],
+                outlierAnalysis: null,
+                cumulative: false,
+            };
+
+            const sharing = {
+                object: {
+                    id: "LEP0WHTGYUe",
+                    name: "name",
+                    publicAccess: "--------",
+                    externalAccess: false,
+                    userGroupAccesses: [{ id: "epFY01iJN0Z", name: "ALL USERS", access: "rw------" }],
+                },
+            };
+
+            let items = [{ id: id }];
+
+            payload.name = name + " - " + id;
+            payload.rows[0].items = rowItems ? items.concat(rowItems.map(id => ({ id: id }))) : items;
+
+            $scope.table = dossiersProgramVisualizationTableFactory.get_table.query(
+                {
+                    filter: "name:eq:" + payload.name,
+                },
+                function (tbl) {
+                    if (tbl && tbl.visualizations[0]) {
+                        if (tbl.visualizations[0].id) {
```

Comment:
[minor] `tbl.visualizations[0].id` used 6 times, so `const visualizationId = tbl.visualizations[0].id`.

Reply:
Done, it should make thing more readable.